### ISSUE TITLE
Git and filesystem auto-refresh slow or not working

### ISSUE THREAD

**Issue by Link512** [microsoft/vscode#82678]
Issue Type: <b>Bug</b>

Ever since the latest major update (1.39) the auto refresh feature for both filesystem and git is either very slow or not working at all.

For GIT:
Each time I edit a file, I have to go to the source control panel and manually click the refresh button for it to pick up the changes I've just made

For filesystem:
If I checkout a new branch, I either have to wait for ~30 seconds or manually hit the refresh button in the Explorer for the changes to show up in the Editor.

VS Code version: Code 1.39.2 (6ab598523be7a800d7f3eb4d92d7ab9a66069390, 2019-10-15T15:33:00.827Z)
OS version: Darwin x64 19.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (4 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|5, 3, 3|
|Memory (System)|16.00GB (0.05GB free)|
|Process Argv|-psn_0_667811|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (24)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.0.5
better-toml|bun|0.3.2
bracket-pair-colorizer|Coe|1.0.61
vscode-markdownlint|Dav|0.31.0
gitlens|eam|10.1.1
EditorConfig|Edi|0.14.2
prettier-vscode|esb|2.3.0
shell-format|fox|6.1.2
terraform|mau|1.4.0
rainbow-csv|mec|1.3.1
vscode-docker|ms-|0.8.1
python|ms-|2019.10.41019
remote-containers|ms-|0.81.0
remote-ssh|ms-|0.47.1
remote-ssh-edit|ms-|0.47.1
remote-ssh-explorer|ms-|0.47.1
remote-wsl|ms-|0.39.9
vscode-remote-extensionpack|ms-|0.17.0
cpptools|ms-|0.26.0
Go|ms-|0.11.7
vsliveshare|ms-|1.0.950
proto|pet|0.0.2
material-icon-theme|PKi|3.9.1
code-settings-sync|Sha|3.4.3

</details>

**Comment by bpasero** [comment]
@Link512 please provide exact steps to reproduce with a repository I can access and steps how to reproduce

**Comment by fbattistella** [comment]
I am running into the same issue, with any and all repositories, github, gitlab, bitbucket....
Steps are easy:
- make changes in a file, save
- no changes show in the source control, so I cant commit the changes

This is not the first time this happens after an update of vscode.

The work around is to command+shift+P and reload the window :(

**Comment by bpasero** [comment]
@fbattistella can you run `code --verbose` and watch the developer tools to see if the file change events are recorded?

**Comment by fbattistella** [comment]
After opening with `code --verbose /path/to/code` I waited for messages in the terminal to stop, changed a file and saved, clicked on the source control icon, no changes, this was the output for that.


[vscode-log.txt](https://github.com/microsoft/vscode/files/3739557/vscode-log.txt)

**Comment by bpasero** [comment]
@fbattistella to see the actual output it has to be looked at in the window itself. Help > Toggle Developer Tools > Console

**Comment by fbattistella** [comment]
hah, forgive my noobness :D will get to that in a sec :D

**Comment by fbattistella** [comment]
did the same steps again, this is the console log.
[-1571320208609.log](https://github.com/microsoft/vscode/files/3739639/-1571320208609.log)

**Comment by fbattistella** [comment]
nope! I guess in my case its addons :) thanks for the help @bpasero !
